### DISCUSSION TITLE
Datadog log integration

### DISCUSSION THREAD

**Discussion by chabou** [vercel/vercel#4007] [General]
Hi all! I am trying to send my api json logs to Datadog with Log Drains. 

The [Now Datadog plugin](https://zeit.co/integrations/datadog-logs) works great but only to retrieve the metrics about the lambda execution (or edge logs). I can't parse the json output of my API lambda (generated with `winston`). It is considered as a long string payload, not a json with some meaningful fields to index. Has anybody successfully achieved this?

Thank you for your help :pray:

**Comment by aecorredor** [comment]
Hey @chabou, I've achieved this with winston by using DataDog's agentless logging: https://docs.datadoghq.com/logs/log_collection/nodejs/?tab=winston30#agentless-logging.

One thing I noticed with using the integration for deployment log drains offered by Zeit is that I was being charged $18/month for an on-demand host. I was wondering if anyone from the Zeit team can confirm that the integration for log drains does indeed incur this monthly charge and that it wasn't me who installed some host by mistake.

**Reply by Noitidart** [reply]
Any developments you had since then? I've been looking into it too. Also for APM did you learn something there? Thanks for sharing!

**Reply by aecorredor** [reply]
Hey, yeah, you'll be charged for a host if you activate the log drain. I ended up disabling and just using plain agentless logging directly from my server code. The downside is that I lost build logs...But you can always inspect those temporarily directly in the Vercel dashboard.

**Reply by Noitidart** [reply]
Thanks for sharing! Was apm possible? Like at the start of every request doing a `require('dd-trace')`?

**Reply by aecorredor** [reply]
I don't think APM actually works if you're running on Vercel serverless functions (since DD needs a host by the container). I think they even have some other topic/issue around this.

**Reply by Noitidart** [reply]
Ah got it thank you!

**Comment by aecorredor** [comment]
While we are at it, has anyone here been able to set up APM (tracing with `dd-trace`)?
